File: desktop/src/com/oni/donjon/desktop/DesktopLauncher.java
package com.oni.donjon.desktop;

import com.badlogic.gdx.Application;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application;
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3ApplicationConfiguration;
import com.oni.donjon.DonjonGame;

public class DesktopLauncher {
    public static void main(String[] arg) {
        Lwjgl3ApplicationConfiguration config = new Lwjgl3ApplicationConfiguration();
        config.setWindowedMode(800, 600);
        new Lwjgl3Application(new DonjonGame(), config);
        Gdx.app.setLogLevel(Application.LOG_DEBUG);
    }
}

File: desktop/src/main/kotlin/com/oni/donjon/desktop/DesktopLauncher.kt
@file:JvmName("DesktopLauncher")

package com.oni.donjon.desktop

import com.badlogic.gdx.Application
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3ApplicationConfiguration
import com.oni.donjon.DonjonGame

fun main() {
    Lwjgl3Application(
        DonjonGame(),
        Lwjgl3ApplicationConfiguration().apply {
            setTitle("Project donjon")
            setWindowedMode(Constants.DEFAULT_WIDTH, Constants.DEFAULT_HEIGHT)
            setResizable(false)
        }
    )
    Gdx.app.logLevel = Application.LOG_DEBUG
}

object Constants {
    const val DEFAULT_WIDTH = 800
    const val DEFAULT_HEIGHT = 600
}
